vm/src/stdlib/math.rs

```diff
@@ -696,6 +697,15 @@ fn math_remainder(x: IntoPyFloat, y: IntoPyFloat, vm: &VirtualMachine) -> PyResu
 
         return Ok(1.0_f64.copysign(x) * r);
     }
+    if x.is_finite() && y.is_infinite() {
```

Comment:
There's currently a lot of duplicate checks going on here and in the following lines until the end of the function. Condensing it down to:

```rust
    if x.is_infinite() && !y.is_nan() {
        return Err(vm.new_value_error("math domain error".to_owned()));
    }
    if x.is_nan() || y.is_nan() {
        return Ok(std::f64::NAN);
    }
    if y.is_infinite() {
        Ok(x)
    } else {
        Err(vm.new_value_error("math domain error".to_owned()))
    }
```

seems to handle the cases fine (at least those tested).

Reply:
Oh,, you're right. (I should have thought that..)
I fixed it thanks to your comment 👍